src/org/zaproxy/zap/extension/pscanrules/SetCookieUtils.java

```diff
@@ -62,6 +65,85 @@ public static boolean hasAttribute(String headerValue, String attributeName) {
         return false;
     }
 
+    /**
+     * Return the value of the specified attribute in the given Set-Cookie header value, or null if it is not present.
+     *
+     * @param headerValue the value of the header
+     * @param attributeName the name of the attribute
+     * @return the attribute value, or null if it is not present
+     */
+    public static String getAttributeValue(String headerValue, String attributeName) {
+        validateParameterNotNull(headerValue, "headerValue");
+        validateParameterNotNull(attributeName, "attributeName");
+
+        if (headerValue.isEmpty() || attributeName.isEmpty()) {
+            return null;
+        }
+
+        String[] cookieElements = headerValue.split(";");
+        if (cookieElements.length == 1 || !isCookieNameValuePairValid(cookieElements[0])) {
+            return null;
+        }
+
+        for (int i = 1; i < cookieElements.length; i++) {
+            String[] attribute = cookieElements[i].split("=", 2);
+            if (attribute.length > 1 && 
+                    attributeName.equalsIgnoreCase(attribute[0].trim())) {
+                return attribute[1].trim();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns the name of the cookie in the given Set-Cookie header value, or null if not found.
+     *
+     * @param cookieHeaderValue the value of the header value
+     * @return the name of the cookie, or null if not found
+     */
+    public static String getCookieName(String cookieHeaderValue) {
+        validateParameterNotNull(cookieHeaderValue, "cookieHeaderValue");
+
+        if (cookieHeaderValue.isEmpty()) {
+            return null;
+        }
+
+        int nameValuePairIdx = cookieHeaderValue.indexOf('=');
+        if (nameValuePairIdx == NOT_FOUND) {
+            return null;
+        }
+
+        return cookieHeaderValue.substring(0, nameValuePairIdx).trim();
+    }
+
+    /**
+     * Returns the relevant SetCookie(2) line as well as just the cookie name, or null if not found.
```

Comment:
What's the significance of `(2)`? Is that meant to indicate that this handles both Set-Cookie and Set-Cookie2? If so I'd suggest being specific here:

> Returns the relevant Set-Cookie or Set-Cookie2 line as well as...

The return statement should mention something like the SetCookie header plus name. Which should be clear enough once the description of the function is updated....my 2cents.

Also you might want to say header vs line, as line would seem to imply the whole header up to some sort of line ending or wordwrap.